src/daemon/default_vm_image_vault.cpp

```diff
@@ -529,6 +533,14 @@ void mp::DefaultVMImageVault::update_images(const FetchType& fetch_type, const P
     for (const auto& key : keys_to_update)
     {
         const auto& record = prepared_image_records[key];
+        {
+            std::lock_guard<decltype(fetch_mutex)> lock{fetch_mutex};
```

Comment:
As discussed, I think the template specification can now be done away with (C++17). That would have to be tested, but hopefully the required deductions guides are in place.